File: examples/governance_static_go/domain/approve_model.go
package domain

import (
	"time"
	"fmt"
)

type ApproveModel struct {
	ModelId string `json:"model_id"`
}

func (c ApproveModel) CommandName() string { return "ApproveModel" }

func (c ApproveModel) Execute(repo AiModelRepository) (*AiModel, *ApprovedModel, error) {
	existing, err := repo.Find(c.ModelId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("AiModel not found: %s", c.ModelId)
	}
	if existing.Status != "classified" {
		return nil, nil, fmt.Errorf("cannot ApproveModel: AiModel is in %s state", existing.Status)
	}
	existing.Status = "approved"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ApprovedModel{
		AggregateID: existing.ID,
		ModelId: c.ModelId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/model_registry_domain/lib/model_registry_domain/ai_model/commands/approve_model.rb
module ModelRegistryDomain
  class AiModel
    module Commands
      class ApproveModel
        include Hecks::Command
        emits "ApprovedModel"

        attr_reader :model_id

        def initialize(model_id: nil)
          @model_id = model_id
        end

        def call
          existing = repository.find(model_id)
          if existing
            unless existing.status == "classified"
              raise Hecks::Error, "Cannot ApproveModel: status must be 'classified', got '#{existing.status}'"
            end
            AiModel.new(
              id: existing.id,
              name: existing.name,
              version: existing.version,
              provider_id: existing.provider_id,
              description: existing.description,
              risk_level: existing.risk_level,
              registered_at: existing.registered_at,
              parent_model_id: existing.parent_model_id,
              derivation_type: existing.derivation_type,
              capabilities: existing.capabilities,
              intended_uses: existing.intended_uses,
              status: "approved"
            )
          else
            raise Hecks::Error, "AiModel not found: #{model_id}"
          end
        end
      end
    end
  end
end
